validator/sawtooth_validator/gossip/gossip.py

```diff
@@ -502,25 +537,32 @@ def _get_peers_of_endpoints(self, peers, endpoints):
 
         for endpoint in endpoints:
             try:
-                if endpoint in self._temp_endpoints:
-                    LOGGER.debug("Endpoint is not ready to be asked about "
-                                 "peers: %s", endpoint)
+                conn_id = self._network.get_connection_id_by_endpoint(
+                    endpoint)
+
+                if conn_id in peers:
+                    # connected and peered - we've already sent
+                    continue
                 else:
-                    conn_id = self._network.get_connection_id_by_endpoint(
-                        endpoint)
-                    if conn_id in peers:
-                        # connected and peered - we've already sent
+                    # connected but not peered
+                    if endpoint in self._temp_endpoints:
+                        LOGGER.debug("Endpoint is not ready to be asked about "
```

Comment:
It would be preferable to log the change of state, not log here where state is not changing (would be less logging to deal with)